dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -46,7 +46,9 @@
         )
       {%- endset -%}
 
-      {{ get_insert_overwrite_merge_sql(target_relation, source_sql, dest_columns, [predicate], include_sql_header=true) }}
+      {{ config.set('include_sql_header', true) }}
+      {{ get_insert_overwrite_merge_sql(target_relation, source_sql, dest_columns, [predicate]) }}
+      {{ config.set('include_sql_header', false) }}
```

Comment:
Oh, this is clever! And does not feel any hackier than what we're doing right now.

Could we add a comment here, to the effect of:
```sql
/* {#
    because we're putting the model SQL _directly_ into the MERGE statement,
    we need to prepend the MERGE statement with the user-configured sql_header,
    which may be needed to resolve that model SQL (e.g. referencing a variable or UDF in the header)
    in the "dynamic" case, we save the model SQL result as a temp table first, wherein the
    sql_header is included by the create_table_as macro
#} */
```